plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,247 @@
+
+      clip_text = copy_messages_to_clipboard(chat_message_3)
+
+      # post transcript in topic
+      topic_page.visit_topic_and_open_composer(topic)
+      topic_page.fill_in_composer(clip_text)
+
+      within ".chat-transcript-messages" do
+        expect(page).to have_content("An interesting article")
+      end
+    end
+
+    it "quotes a single chat message into another chat message " do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      # select message + copy to clipboard
+      clip_text = copy_messages_to_clipboard(chat_message_5)
+      click_selection_button("cancel")
+
+      # send transcript message in chat
+      chat_channel_page.fill_composer(clip_text)
+      chat_channel_page.click_send_message
+      message = nil
+      try_until_success do
+        message = ChatMessage.find_by(user: current_user, message: clip_text.chomp)
+        expect(message).not_to eq(nil)
+      end
+      expect(chat_channel_page).to have_message(id: message.id)
+      within chat_channel_page.message_by_id(message.id) do
+        expect(page).to have_css(".chat-transcript")
+      end
+    end
+  end
+
+  describe "quoting into a topic directly" do
+    it "opens the topic composer with the quote prefilled and the channel category preselected" do
+      topic.update!(category: chat_channel_1.chatable)
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      # select message + prefill in composer
+      select_message(chat_message_5)
+      click_selection_button("quote")
+      expect(topic_page).to have_expanded_composer
+      expect(topic_page).to have_composer_content(generate_transcript(chat_message_5, current_user))
+      expect(page).to have_css(
+        ".category-input .select-kit-header[data-value='#{chat_channel_1.chatable.id}']",
+      )
+      expect(page).not_to have_current_path(chat_channel_1.chatable.url)
+
+      # create the topic with the transcript as the OP
+      topic_page.fill_in_composer_title("Some topic title for testing")
+      topic_page.send_reply
+      expect(page).to have_content("Some topic title for testing")
+      topic = Topic.where(user: current_user).last
+      expect(page).to have_current_path(topic.url)
+      within topic_page.post_by_number(1) do
+        expect(page).to have_css(".chat-transcript")
+      end
+
+      # ensure the transcript date is formatted
+      expect(page).to have_css(".chat-transcript-datetime a[data-date-formatted=\"true\"]")
+    end
+
+    context "when on mobile" do
+      it "first navigates to the channel's category before opening the topic composer with the quote prefilled",
+         mobile: true do
+        topic.update!(category: chat_channel_1.chatable)
```

Comment:
I prefer prefer block for this, it separates initial state from actual user actions